ambry-cloud/src/test/java/com.github.ambry.cloud/CloudBlobStoreTest.java

```diff
@@ -614,4 +621,196 @@ private BlobId getUniqueId(short accountId, short containerId, boolean encrypted
+    testGetWithAtleastOneNonExistentBlob(singleIdList);
+
+    //test get with one blob in list non-existent
+    blobIds.add(nonExistentId);
+    testGetWithAtleastOneNonExistentBlob(blobIds);
+    blobIds.remove(blobIds.remove(blobIds.size() - 1));
+
+    //test get for deleted blob
+    MessageInfo deletedMessageInfo = messageWriteSet.getMessageSetInfo().get(0);
+    MockMessageWriteSet deleteMockMessageWriteSet = new MockMessageWriteSet();
+    deleteMockMessageWriteSet.add(deletedMessageInfo, null);
+    store.delete(deleteMockMessageWriteSet);
+
+    //test get with a deleted blob in blob list to get, but {@code StoreGetOptions.Store_Include_Deleted} not set in get options
+    testGetForDeletedBlobWithoutIncludeDeleteOption(blobIds);
+
+    //test get with a deleted blob in blob list to get, and {@code StoreGetOptions.Store_Include_Deleted} set in get options
+    testGetForDeletedBlobWithIncludeDeleteOption(blobIds, (BlobId) deletedMessageInfo.getStoreKey());
+
+    blobIds.remove(0);
+
+    //test get for expired blob
+    BlobId expiredBlobId = forceUploadExpiredBlob();
+    blobIds.add(expiredBlobId);
+
+    //test get with an expired blob in blob list, but {@code StoreGetOptions.Store_Include_Expired} not set in get options
+    testGetForDeletedBlobWithoutIncludeExpiredOption(blobIds);
+
+    //test get with a expired blob in blob list to get, and {@code StoreGetOptions.Store_Include_Expired} set in get options
+    testGetForDeletedBlobWithIncludeExpiredOption(blobIds, expiredBlobId);
+  }
+
+  /**
+   * Test cloud store get with a list of blobs that are all valid and previously uploaded in the store
+   * @param blobIds list of blob ids to get
+   * @param blobIdToUploadedDataMap map of expected blobid to data buffers
+   * @throws Exception
+   */
+  private void testGetForExistingBlobs(List<BlobId> blobIds, Map<BlobId, ByteBuffer> blobIdToUploadedDataMap)
+      throws Exception {
+    StoreInfo storeInfo = store.get(blobIds, EnumSet.noneOf(StoreGetOptions.class));
+    assertTrue(
+        "Number of records returned by get should be " + blobIds.size() + " found " + storeInfo.getMessageReadSetInfo()
+            .size(), storeInfo.getMessageReadSetInfo().size() == blobIds.size());
+    for (int i = 0; i < storeInfo.getMessageReadSetInfo().size(); i++) {
+      if (blobIdToUploadedDataMap.containsKey(storeInfo.getMessageReadSetInfo().get(i).getStoreKey())) {
+        ByteBuffer uploadedData = blobIdToUploadedDataMap.get(storeInfo.getMessageReadSetInfo().get(i).getStoreKey());
+        ByteBuffer downloadedData = ByteBuffer.allocate((int) storeInfo.getMessageReadSetInfo().get(i).getSize());
+        WritableByteChannel writableByteChannel = Channels.newChannel(new ByteBufferOutputStream(downloadedData));
+        storeInfo.getMessageReadSet().writeTo(i, writableByteChannel, -1, -1);
+        downloadedData.flip();
+        assertTrue(uploadedData.equals(downloadedData));
```

Comment:
All the places where you call assertTrue(expected.equals(actual)), better to use assertEquals("error message", expected, actual)